### PR TITLE
Feature/jhbiggs/1581 remove dagger android

### PR DESCRIPTION
**Pull request by jhbiggs**
#1581 and it passed githooks this time.  Thanks @macgills for your tireless patience and assistance!

### CODE REVIEW

File: app/src/main/java/org/kiwix/kiwixmobile/local_file_transfer/LocalFileTransferActivity.java [hunk 1]
```diff
@@ -99,12 +102,9 @@
 
   private List<WifiP2pDevice> availablePeerDevices = new ArrayList<WifiP2pDevice>();
 
-  @Override protected void injection() {
-    ActivityExtensionsKt.getKiwixActivityComponent(this).inject(this);
-  }
-
   @Override
   protected void onCreate(@Nullable Bundle savedInstanceState) {
+    ActivityExtensionsKt.getKiwixActivityComponent(this).inject(this);
```
Comment (macgills) [inline app/src/main/java/org/kiwix/kiwixmobile/local_file_transfer/LocalFileTransferActivity.java]: Double injection

File: app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt [hunk 2]
```diff
@@ -48,11 +49,8 @@ import java.io.File
 
 class KiwixMainActivity : CoreMainActivity() {
 
-  override fun injection() {
-    kiwixActivityComponent.inject(this)
-  }
-
   override fun onCreate(savedInstanceState: Bundle?) {
+    kiwixActivityComponent.inject(this)
```
Comment (macgills) [inline app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt]: double injection

File: app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt [hunk 3]
```diff
@@ -104,6 +102,10 @@ class KiwixMainActivity : CoreMainActivity() {
     }
   }
 
+  override fun injection(coreComponent: CoreComponent) {
+    this.kiwixActivityComponent.inject(this)
```
Comment (macgills) [inline app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt]: unnecessary `this`

File: app/src/main/java/org/kiwix/kiwixmobile/zim_manager/ZimManageActivity.kt [hunk 4]
```diff
@@ -41,6 +42,9 @@ import org.kiwix.kiwixmobile.local_file_transfer.LocalFileTransferActivity
 import javax.inject.Inject
 
 class ZimManageActivity : BaseActivity() {
+  override fun injection(coreComponent: CoreComponent) {
+    kiwixActivityComponent.inject(this)
```
Comment (macgills) [inline app/src/main/java/org/kiwix/kiwixmobile/zim_manager/ZimManageActivity.kt]: this should be `cachedComponent.inject(this)`, I needed to store the activityComponent in a field on this object because other objects need access to the component to inject their fields

File: core/src/main/java/org/kiwix/kiwixmobile/core/CoreApp.java [hunk 5]
```diff
@@ -63,6 +60,8 @@ public static CoreComponent getCoreComponent() {
     return coreComponent;
   }
 
+  public static ActivityComponent getActivityComponent() { return activityComponent; }
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/CoreApp.java]: I believe this is unused because there is no set method, if you were to store an ActivityComponent on the Application object you would 100% be causing a memory leak

File: core/src/main/java/org/kiwix/kiwixmobile/core/di/modules/ActivityModule.kt [hunk 6]
```diff
@@ -35,13 +35,15 @@ import org.kiwix.kiwixmobile.core.utils.AlertDialogShower
...
 
 @Module
-abstract class ActivityModule {
-  @Binds
+class ActivityModule {
+
+  @Provides
   @ActivityScope
-  abstract fun bindDialogShower(alertDialogShower: AlertDialogShower): DialogShower
+  internal fun providesDialogShower(activity: Activity): DialogShower = AlertDialogShower(activity)
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/di/modules/ActivityModule.kt]: revert this change, an abstract class with an `@Binds` method didn't need changing

File: gradle.properties [hunk 7]
```diff
@@ -1,6 +1,17 @@
...
+# This option should only be used with decoupled projects. More details, visit
+# http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
+# org.gradle.parallel=true
+#Sat Jan 11 17:55:48 CST 2020
 android.enableJetifier=true
+android.enableD8.desugaring=true
 android.useAndroidX=true
-org.gradle.jvmargs=-Xmx4096m
-
+android.enableD8=true
```
Comment (macgills) [inline gradle.properties]: revert changes to this file, our build is failing with OOM errors

File: core/src/main/java/org/kiwix/kiwixmobile/core/di/components/ActivityComponent.kt [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+import org.kiwix.kiwixmobile.core.di.modules.ActivityModule
+import org.kiwix.kiwixmobile.core.di.modules.DialogModule
+
+@ActivityScope
+@Subcomponent(
+  modules = [
+    ActivityModule::class,
+    DialogModule::class]
+)
+interface ActivityComponent {
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/di/components/ActivityComponent.kt]: I think this component is unused and can be deleted

File: app/src/main/java/org/kiwix/kiwixmobile/intro/IntroModule.kt [hunk 9]
```diff
@@ -19,10 +19,10 @@ package org.kiwix.kiwixmobile.intro
 
 import dagger.Binds
 import dagger.Module
-import org.kiwix.kiwixmobile.core.di.ActivityScope
 
 @Module
 abstract class IntroModule {
-  @ActivityScope @Binds
+  // @ActivityScope
```
Comment (macgills) [inline app/src/main/java/org/kiwix/kiwixmobile/intro/IntroModule.kt]: Delete comment

File: app/src/main/java/org/kiwix/kiwixmobile/intro/IntroPresenter.java [hunk 10]
```diff
@@ -23,7 +23,7 @@
 import org.kiwix.kiwixmobile.core.di.ActivityScope;
 import org.kiwix.kiwixmobile.core.utils.SharedPreferenceUtil;
 
-@ActivityScope
+//@ActivityScope
```
Comment (macgills) [inline app/src/main/java/org/kiwix/kiwixmobile/intro/IntroPresenter.java]: delete comment

File: core/src/main/java/org/kiwix/kiwixmobile/core/di/components/ActivityComponent.kt (hunk 8 above)
Comment (macgills) [reply core/src/main/java/org/kiwix/kiwixmobile/core/di/components/ActivityComponent.kt]: This will probably no longer be the case when you merge with my changes to SearchActivity

File: app/src/main/java/org/kiwix/kiwixmobile/intro/IntroActivity.java [hunk 11]
```diff
@@ -79,7 +81,8 @@ public void onPageScrollStateChanged(int state) {
     };
   private View[] views;
 
-  @Override protected void injection() {
+  @Override protected void injection(
```
Comment (macgills) [inline app/src/main/java/org/kiwix/kiwixmobile/intro/IntroActivity.java]: this method declaration can be one line, no need for the break I would think

File: core/src/main/java/org/kiwix/kiwixmobile/core/error/ErrorActivity.java [hunk 12]
```diff
@@ -181,4 +181,9 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
     super.onActivityResult(requestCode, resultCode, data);
     restartApp();
   }
+
+  @Override protected void injection(
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/error/ErrorActivity.java]: CoreComponent doesn't need to be on a newline

File: core/src/main/java/org/kiwix/kiwixmobile/core/di/components/ActivityComponent.kt (hunk 8 above)
Comment (macgills) [reply core/src/main/java/org/kiwix/kiwixmobile/core/di/components/ActivityComponent.kt]: @jhbiggs this is the only substantial thing left to do, merge develop into your branch and resolve the conflicts.

File: core/src/main/java/org/kiwix/kiwixmobile/core/search/viewmodel/effects/ShowDeleteSearchDialog.kt [hunk 13]
```diff
@@ -36,7 +35,6 @@ data class ShowDeleteSearchDialog(
   @Inject lateinit var dialogShower: DialogShower
 
   override fun invokeWith(activity: AppCompatActivity) {
-    (activity as SearchActivity).activityComponent.inject(this)
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/search/viewmodel/effects/ShowDeleteSearchDialog.kt]: This needs the injection, it is why I retrieved the activity component from the activity. I can explain in more depth if you want

File: core/src/main/java/org/kiwix/kiwixmobile/core/settings/CoreSettingsActivity.java [hunk 14]
```diff
@@ -70,4 +72,8 @@ private void setUpToolbar() {
 
     toolbar.setNavigationOnClickListener(v -> onBackPressed());
   }
+
+  @Override protected void injection(CoreComponent coreComponent) {
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/settings/CoreSettingsActivity.java]: abstract activity classes don't need to perform injection, this can be deleted

**Comment by jhbiggs** [comment]
@macgills I made the changes above.  The local test failed, however, but not because of a file I had changed.  It appeared to be due to "file_management_no_files" view not found in zim_list.xml.  I am committing therefore despite the failed test.

**Comment by macgills** [comment]
The local tests depend on the app being a fresh install with no extra zim files. Anyway I restarted the checks on this PR, on the emulator the tests are a bit flaky sadly
